prov/usnic/src/usdf_cq.c

```diff
@@ -954,9 +716,9 @@ static struct fi_ops_cq usdf_cq_context_ops = {
 static struct fi_ops_cq usdf_cq_context_soft_ops = {
 	.size = sizeof(struct fi_ops_cq),
 	.read = usdf_cq_read_context_soft,
-	.readfrom = usdf_cq_readfrom_context_soft,
+	.readfrom = usdf_cq_readfrom_context,
```

Comment:
> Returning -FI_ENOSYS is neither an increase nor decrease in functionality, so I don't know how much it matters, but complying w/ the man page seemed like the right thing to do as a tie breaker.

Right, but it would get confusing if the endpoint is opened with `FI_SOURCE` and this gets called and we place `FI_ADDR_NOTAVAIL` in the source address. It's not the information isn't available, it's that we haven't implemented the function.